### ISSUE TITLE
🐛 Expected a block mapping entry false positive

### ISSUE THREAD

**Issue by Netail** [biomejs/biome#11863]
A Yaml map with 2 of the same keys + a comment (not without the comment nor only unique keys) crashes and expects a block mapping entry

Input
```
# should not generate diagnostics
person:
  #·biome-ignore·lint/nursery/noDuplicateMapKeys: test
  name: John Doe
  age: 50
  name: Jane Doe
```

Output
```
× Expected a block mapping entry but instead found 'age: 50
      name: Jane Doe'.
  
    4 │   # biome-ignore lint/nursery/noDuplicateMapKeys: <explanation>
    5 │     name: John Doe
  > 6 │   age: 50
      │   ^^^^^^^
  > 7 │   name: Jane Doe
      │   ^^^^^^^^^^^^^^
    8 │ 
  
  i Expected a block mapping entry here.
  
    4 │   # biome-ignore lint/nursery/noDuplicateMapKeys: <explanation>
    5 │     name: John Doe
  > 6 │   age: 50
      │   ^^^^^^^
  > 7 │   name: Jane Doe
      │   ^^^^^^^^^^^^^^
    8 │ 
  


Parsed tree:
0: YAML_ROOT@0..210
  0: YAML_DOCUMENT_LIST@0..210
    0: YAML_DOCUMENT@0..210
      0: (empty)
      1: YAML_DIRECTIVE_LIST@0..0
      2: (empty)
      3: YAML_BLOCK_IN_BLOCK_NODE@0..210
        0: YAML_PROPERTY_LIST@0..0
        1: YAML_BLOCK_MAPPING@0..210
          0: MAPPING_START@0..34 "" [Comments("# should not generate ..."), Newline("\n")] []
          1: YAML_BLOCK_MAP_ENTRY_LIST@34..208
            0: YAML_BLOCK_MAP_IMPLICIT_ENTRY@34..181
              0: YAML_FLOW_YAML_NODE@34..40
                0: YAML_PROPERTY_LIST@34..34
                1: YAML_PLAIN_SCALAR@34..40
                  0: PLAIN_LITERAL@34..40 "person" [] []
              1: COLON@40..41 ":" [] []
              2: YAML_BLOCK_IN_BLOCK_NODE@41..181
                0: YAML_PROPERTY_LIST@41..41
                1: YAML_BLOCK_MAPPING@41..181
                  0: MAPPING_START@41..167 "" [Newline("\n"), Whitespace("  "), Comments("#·biome-ignore·lint ..."), Newline("\n"), Whitespace("  "), Comments("# biome-ignore lint/n ..."), Newline("\n"), Whitespace("    ")] []
                  1: YAML_BLOCK_MAP_ENTRY_LIST@167..181
                    0: YAML_BLOCK_MAP_IMPLICIT_ENTRY@167..181
                      0: YAML_FLOW_YAML_NODE@167..171
                        0: YAML_PROPERTY_LIST@167..167
                        1: YAML_PLAIN_SCALAR@167..171
                          0: PLAIN_LITERAL@167..171 "name" [] []
                      1: COLON@171..173 ":" [] [Whitespace(" ")]
                      2: YAML_FLOW_IN_BLOCK_NODE@173..181
                        0: FLOW_START@173..173 "" [] []
                        1: YAML_FLOW_YAML_NODE@173..181
                          0: YAML_PROPERTY_LIST@173..173
                          1: YAML_PLAIN_SCALAR@173..181
                            0: PLAIN_LITERAL@173..181 "John Doe" [] []
                        2: FLOW_END@181..181 "" [] []
                  2: MAPPING_END@181..181 "" [] []
            1: YAML_BOGUS_BLOCK_MAP_ENTRY@181..208
              0: MAPPING_START@181..184 "" [Newline("\n"), Whitespace("  ")] []
              1: PLAIN_LITERAL@184..187 "age" [] []
              2: COLON@187..189 ":" [] [Whitespace(" ")]
              3: FLOW_START@189..189 "" [] []
              4: PLAIN_LITERAL@189..191 "50" [] []
              5: FLOW_END@191..191 "" [] []
              6: PLAIN_LITERAL@191..198 "name" [Newline("\n"), Whitespace("  ")] []
              7: COLON@198..200 ":" [] [Whitespace(" ")]
              8: FLOW_START@200..200 "" [] []
              9: PLAIN_LITERAL@200..208 "Jane Doe" [] []
              10: FLOW_END@208..208 "" [] []
              11: MAPPING_END@208..208 "" [] []
          2: MAPPING_END@208..210 "" [Newline("\n"), Newline("\n")] []
      4: (empty)
  1: EOF@210..210 "" [] []

Printed tree:
# should not generate diagnostics
person:
  #·biome-ignore·lint/nursery/noDuplicateMapKeys: test
  # biome-ignore lint/nursery/noDuplicateMapKeys: <explanation>
    name: John Doe
  age: 50
  name: Jane Doe
```

**Comment by ematipico** [comment]
@Netail please provide a playground link

**Comment by ematipico** [comment]
Looking closer to the diagnostic, it's clear that issue isn't the parser, but the action itself. The suppression action added an indentation to the input , causing a parse  error 

This passes

https://biomejs.dev/playground/?lintRules=useExhaustiveSwitchCases&view=syntax&language=yaml#code=IwAgAHMAaABvAHUAbABkACAAbgBvAHQAIABnAGUAbgBlAHIAYQB0AGUAIABkAGkAYQBnAG4AbwBzAHQAaQBjAHMACgBwAGUAcgBzAG8AbgA6AAoAIAAgACMAtwBiAGkAbwBtAGUALQBpAGcAbgBvAHIAZQAgAGwAaQBuAHQALwBuAHUAcgBzAGUAcgB5AC8AbgBvAEQAdQBwAGwAaQBjAGEAdABlAE0AYQBwAEsAZQB5AHMAOgAgAHQAZQBzAHQACgAgACAAbgBhAG0AZQA6ACAASgBvAGgAbgAgAEQAbwBlAAoAIAAgAGEAZwBlADoAIAA1ADAACgAgACAAbgBhAG0AZQA6ACAASgBhAG4AZQAgAEQAbwBlAA%3D%3D

This is what the action emits
https://biomejs.dev/playground/?lintRules=useExhaustiveSwitchCases&view=syntax&language=yaml#code=IwAgAHMAaABvAHUAbABkACAAbgBvAHQAIABnAGUAbgBlAHIAYQB0AGUAIABkAGkAYQBnAG4AbwBzAHQAaQBjAHMACgBwAGUAcgBzAG8AbgA6AAoAIAAgACMAtwBiAGkAbwBtAGUALQBpAGcAbgBvAHIAZQAgAGwAaQBuAHQALwBuAHUAcgBzAGUAcgB5AC8AbgBvAEQAdQBwAGwAaQBjAGEAdABlAE0AYQBwAEsAZQB5AHMAOgAgAHQAZQBzAHQACgAgACAAIAAgAG4AYQBtAGUAOgAgAEoAbwBoAG4AIABEAG8AZQAKACAAIABhAGcAZQA6ACAANQAwAAoAIAAgAG4AYQBtAGUAOgAgAEoAYQBuAGUAIABEAG8AZQA%3D

**Comment by dyc3** [comment]
So its still a bug, why close the issue?

**Comment by ematipico** [comment]
Because this isn't implemented? Netail is working on the YAML analyser

**Comment by dyc3** [comment]
Maybe he wanted this as a todo item, separate from setting up the analyzer. It might be complicated enough to warrant a different PR. Just a guess though. I'll let Netail clarify.

**Comment by Netail** [comment]
Hi, yes it was meant as a kind of todo for myself to finish the yaml analyzer PR. (forgot to assign myself) Didn't investigate that much yet where the parsing issue came from so it wasn't sure if it actually came from the analyzer changes
